trpc/queries/goalV2.ts

```diff
@@ -282,3 +282,47 @@ export const goalBaseQuery = () => {
             sql`to_json("State")`.as('state'),
         ]);
 };
+
+export const getDeepParentGoalIds = (ids: string[]) => {
+    return db
+        .withRecursive('parentsTree', (qb) =>
+            qb
+                .selectFrom('GoalAchieveCriteria')
+                .select('GoalAchieveCriteria.goalId as id')
+                .where('GoalAchieveCriteria.criteriaGoalId', 'is not', null)
```

Comment:
I think that condition is useless, 'cause you filtering select by containing in ids list on the next line

Also, this may be applied in `union` statement